compiler/rustc_mir_transform/src/errors.rs

```diff
@@ -138,13 +140,20 @@ impl<'a> DecorateLint<'a, ()> for UnsafeOpInUnsafeFn {
         self,
         diag: &'b mut DiagnosticBuilder<'a, ()>,
     ) -> &'b mut DiagnosticBuilder<'a, ()> {
-        let desc = diag
-            .handler()
-            .expect("lint should not yet be emitted")
-            .eagerly_translate_to_string(self.details.label(), [].into_iter());
+        let handler = diag.handler().expect("lint should not yet be emitted");
+        let desc = handler.eagerly_translate_to_string(self.details.label(), [].into_iter());
         diag.set_arg("details", desc);
         diag.span_label(self.details.span, self.details.label());
         diag.note(self.details.note());
+
+        if let Some((start, end)) = self.suggest_unsafe_block {
+            diag.multipart_suggestion_verbose(
+                crate::fluent_generated::_subdiag::suggestion,
```

Comment:
Ah, also I assume using `_subdiag` here is wrong, but stuff around the fluent messages seems to have changed compared to the rustc-dev-guide, so I wasn't able to figure out how to access the subdiagnostics correctly.

Reply:
Found the solution by checking the macro impl, it adds the crate prefix onto the labels.